File: app/src/main/java/com/example/furnitureworld/BrowseFurnitureActivity.java
package com.example.furnitureworld;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.firebase.ui.database.FirebaseRecyclerOptions;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

public class BrowseFurnitureActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private Adaptor adaptor;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_browse_furniture);

        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        DatabaseReference furnitureRef = FirebaseDatabase.getInstance().getReference("furniture");

        FirebaseRecyclerOptions<Furniture> options =
                new FirebaseRecyclerOptions.Builder<Furniture>()
                        .setQuery(furnitureRef, Furniture.class)
                        .build();

        adaptor = new Adaptor(options);
        recyclerView.setAdapter(adaptor);
    }

    @Override
    protected void onStart() {
        super.onStart();
        adaptor.startListening();
    }

    @Override
    protected void onStop() {
        super.onStop();
        adaptor.stopListening();
    }
}

File: app/src/main/java/com/app/furnitureworld/BrowseFurnitureActivity.kt
package com.app.furnitureworld

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.app.furnitureworld.adaptors.BrowseItemsAdapter
import kotlinx.android.synthetic.main.activity_browse_items.*

class BrowseFurnitureActivity : AppCompatActivity(), BrowseItemsAdapter.BrowseListener {

    var list: MutableList<String> = ArrayList<String>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_browse_items)



        rv_browse_items.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL , false)
        val adapter = BrowseItemsAdapter(this@BrowseFurnitureActivity, list, this)
        rv_browse_items.adapter = adapter
    }

    override fun onBrowseClick(pos: Int, action: String) {

    }
}